broker/proto/neon_broker.rs

```diff
@@ -0,0 +1,383 @@
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct SubscribeSafekeeperInfoRequest {
+    #[prost(oneof="subscribe_safekeeper_info_request::SubscriptionKey", tags="1, 2")]
+    pub subscription_key: ::core::option::Option<subscribe_safekeeper_info_request::SubscriptionKey>,
+}
+/// Nested message and enum types in `SubscribeSafekeeperInfoRequest`.
+pub mod subscribe_safekeeper_info_request {
+    #[derive(Clone, PartialEq, ::prost::Oneof)]
+    pub enum SubscriptionKey {
+        /// subscribe to everything
+        #[prost(message, tag="1")]
+        All(super::Empty),
+        /// subscribe to specific timeline
+        #[prost(message, tag="2")]
+        TenantTimelineId(super::TenantTimelineId),
+    }
+}
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct SafekeeperTimelineInfo {
+    #[prost(uint64, tag="1")]
+    pub safekeeper_id: u64,
+    #[prost(message, optional, tag="2")]
+    pub tenant_timeline_id: ::core::option::Option<TenantTimelineId>,
+    /// Term of the last entry.
+    #[prost(uint64, tag="3")]
+    pub last_log_term: u64,
+    /// LSN of the last record.
+    #[prost(uint64, tag="4")]
+    pub flush_lsn: u64,
+    /// Up to which LSN safekeeper regards its WAL as committed.
+    #[prost(uint64, tag="5")]
+    pub commit_lsn: u64,
+    /// LSN up to which safekeeper has backed WAL.
+    #[prost(uint64, tag="6")]
+    pub backup_lsn: u64,
+    /// LSN of last checkpoint uploaded by pageserver.
+    #[prost(uint64, tag="7")]
+    pub remote_consistent_lsn: u64,
+    #[prost(uint64, tag="8")]
+    pub peer_horizon_lsn: u64,
+    #[prost(uint64, tag="9")]
+    pub local_start_lsn: u64,
+    /// A connection string to use for WAL receiving.
+    #[prost(string, tag="10")]
+    pub safekeeper_connstr: ::prost::alloc::string::String,
+}
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct TenantTimelineId {
+    #[prost(bytes="vec", tag="1")]
+    pub tenant_id: ::prost::alloc::vec::Vec<u8>,
+    #[prost(bytes="vec", tag="2")]
+    pub timeline_id: ::prost::alloc::vec::Vec<u8>,
+}
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct Empty {
+}
+/// Generated client implementations.
```

Comment:
Curious, how it gets [re]generated? Could add some module docs for that.

Reply:
Ouch, I forgot to add this file to .gitignore :) build.rs generates it.